### PR TITLE
Add service info retrieving [ECR-3859]

### PR DESCRIPTION
**Pull request by MakarovS**
## Overview
Add service info retrieving.
Targeted into `ECR-3860` until it's merged.

---
See: https://jira.bf.local/browse/ECR-3859

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

### CODE REVIEW

File: exonum-light-client/README.md [hunk 1]
```diff
@@ -121,7 +121,18 @@ if available, transaction location and result:
 Optional<TransactionResponse> response = exonumClient.getTransaction(txHash);
 ```
 * `txHash` is a hash of the transaction to search.
-  
+
+### Retrieving service info
+To build a transaction, service id is needed. It can be obtained with either
```
Comment (dmitry-timofeev) [inline exonum-light-client/README.md]: This documentation seems out of place. I think it shall be included in the section on creating a transaction message, saying that the service id, required to create a message, can be obtained, if needed, by the service name:
```java
// Update the code to be correct and compile!
int serviceId =  exonumClient.getServiceInfoByName(serviceName)
    .mapToInt(#getId)
    .orElseThrow(() -> new IllegalStateException("No service with the given name found: " + serviceName);
```

This section may remain, but rewritten as just the way to get information on active services in the blockchain.

File: exonum-light-client/src/main/java/com/exonum/client/ExonumClient.java [hunk 2]
```diff
@@ -183,6 +184,19 @@ BlocksRange getLastBlocks(int size, BlockFilteringOption blockFilter,
    */
   Optional<Block> getLastNonEmptyBlock();
 
+  /**
+   * Returns the service info of a started service instance with a given name; or
+   * {@code Optional.empty()} if there is no service instance with such name.
+   *
+   * @param serviceName the name of a service instance
+   */
+  Optional<ServiceInfo> getServiceInfoByName(String serviceName);
```
[the remark this answers is not in the document]
Comment (MakarovS) [reply exonum-light-client/src/main/java/com/exonum/client/ExonumClient.java]: Changed to `findServiceInfoByName`

File: exonum-light-client/README.md (hunk 1 above)
Comment (dmitry-timofeev) [reply exonum-light-client/README.md]: > but rewritten as just *the way to get information on active services in the blockchain*.

It still starts as "To build a transaction, service id is needed."

File: exonum-light-client/src/main/java/com/exonum/client/ExonumClient.java [hunk 3]
```diff
@@ -190,10 +190,12 @@ BlocksRange getLastBlocks(int size, BlockFilteringOption blockFilter,
    *
    * @param serviceName the name of a service instance
    */
-  Optional<ServiceInfo> getServiceInfoByName(String serviceName);
+  Optional<ServiceInfo> findServiceInfoByName(String serviceName);
```
Comment (dmitry-timofeev) [inline exonum-light-client/src/main/java/com/exonum/client/ExonumClient.java]: Maybe drop `ByName` since it must be clear enough from the parameter list and there are no other String query parameters (i.e., no `BySomeOtherStringProperty` that won't work with overloads)?

File: exonum-light-client/src/test/java/com/exonum/client/ExplorerApiHelperTest.java [hunk 4]
```diff
@@ -232,5 +234,8 @@ void parseServicesResponse() {
 
     List<ServiceInfo> actual = ExplorerApiHelper.parseServicesResponse(json);
     assertThat(actual, contains(expected.toArray()));
+    int serviceId = Optional.of(serviceInfo1)
```
Comment (dmitry-timofeev) [inline exonum-light-client/src/test/java/com/exonum/client/ExplorerApiHelperTest.java]: That must be a draft.

File: exonum-light-client/src/test/java/com/exonum/client/ExplorerApiHelperTest.java [hunk 5]
```diff
@@ -232,5 +234,8 @@ void parseServicesResponse() {
...
+        .map(ServiceInfo::getId)
```
Comment (dmitry-timofeev) [inline exonum-light-client/src/test/java/com/exonum/client/ExplorerApiHelperTest.java]: Will `mapToInt` be more appropriate (in the example also)? OptionalInt also has https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/OptionalInt.html#orElseThrow(java.util.function.Supplier)

Comment (MakarovS) [reply exonum-light-client/src/test/java/com/exonum/client/ExplorerApiHelperTest.java]: Sorry, either I don't understand what exactly do you propose here or it's something like
```
    int serviceId = Optional.of(serviceInfo1).stream()
        .mapToInt(ServiceInfo::getId)
        .findFirst()
        .orElseThrow(() -> new IllegalStateException("No service with the given name found: " + serviceName);
```
which doesn't seem better to me.
`Optional` doesn't have `mapToInt` method.

Comment (dmitry-timofeev) [reply exonum-light-client/src/test/java/com/exonum/client/ExplorerApiHelperTest.java]: I see, sorry, I must have missed there is no mapToInt in Optional. It's good then, a stream is certainly inferior.

File: exonum-light-client/src/test/java/com/exonum/client/ExplorerApiHelperTest.java [hunk 6]
```diff
@@ -194,8 +193,42 @@ void parseServicesResponse() {
 
     List<ServiceInfo> actual = ExplorerApiHelper.parseServicesResponse(json);
     assertThat(actual, contains(expected.toArray()));
-    int serviceId = Optional.of(serviceInfo1)
-        .map(ServiceInfo::getId)
-        .orElseThrow(() -> new IllegalStateException("No service with the given name found: " + serviceName);
+  }
+
+  private static Stream<Arguments> testData() {
```
Comment (dmitry-timofeev) [inline exonum-light-client/src/test/java/com/exonum/client/ExplorerApiHelperTest.java]: That seems unrelated.

**Comment by dmitry-timofeev** [comment]
Why does it target ECR-3860 and not `master`?

**Comment by MakarovS** [comment]
> Why does it target ECR-3860 and not master?

Sorry, I expected ECR-3860 to be merged before this one and it was easier to develop based on a working version of LC.
